Remove gulp and add tasks with simple npm tasks

oliverguenther opened the pull request:
This removes `gulp` and provides a simpler `karma` and `webpack`,
`webpack-watch` npm scripts.

For actual development, users will need the Rails stack anyway, so it
won't make sense to keep a separate set of dependencies up to date.
- Instead of `gulp webpack`, use `npm run webpack`.
- Instead of `gulp watch`, use `npm run webpack-watch`, which is _drastically_ faster than gulp watch-polling.
- Instead of `gulp tests:karma` use `npm run karma`, or just `npm test`.

This also removes all traces of express and restores the Rails styleguide. It is not whitelisted for precompilation and is only available in the context of the development mode.
To access the styleguide, use `localhost:5000/styleguide`, which is now rendered through the Rails stack.

It also replaces the unicorn web server in the `Procfile.dev, which causes code reloading for asset requests and is terribly slow compared to thin in that context.

/cc @opf/frontend

oliverguenther committed:
Add redirect for /styleguide

Adds a development-only route for `/styleguide` as an alias to
`/assets/styleguide.html`.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-housekeeping-remove-gulp.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-housekeeping-remove-gulp.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Bump livingstyleguide

Avoids redcarpet-3.0.0 XSS
https://github.com/livingstyleguide/livingstyleguide/issues/161

oliverguenther committed:
Remove gulp with simpler npm tasks

This removes `gulp` and provides a simpler `karma` and `webpack`,
`webpack-watch` npm scripts.

For actual development, users will need the Rails stack anyway, so it
won't make sense to keep a separate set of dependencies up to date.

* Instead of `gulp webpack`, use `npm run webpack`.
* Instead of `gulp watch`, use `npm run webpack-watch`, which is *drastically* faster than gulp watch-polling.
* Instead of `gulp tests:karma` use `npm run karma`, or just `npm test`.

This also removes all traces of express and restores the Rails
styleguide. It is not whitelisted for precompilation and is only
available in the context of the development mode.

To access the styleguide, use `/assets/styleguide.html`.
